### PR TITLE
Fixed decimals for non-standard networks

### PR DESCRIPTION
**Pull request by evmosius**
TLDR: Added an optional `decimals` field to network.json, decimals default to 1e6 unless it is overridden (in the case of evmos 1e18).

Unsure of how fix the error below, some insight would be appreciated.

[ ] Fix `Failed to broadcast: Unsupported type: '/ethermint.types.v1.EthAccount'` Error
Screen Shot 2022-03-06 at 10 48 55 AM

### CODE REVIEW

**Comment by tombeynon** [comment]
This is incredible, thanks so much for doing this. Desperately want to get Evmos onto the site but been juggling other fixes etc this weekend 💥 

I've just looked into the unsupported type. I believe the solution will be to use this repository to provide the additional message types we need to handle Evmos transaction. There is a Delegate message builder which is the main thing we need. 

https://github.com/tharsis/evmosjs

I can probably spend some time on this tomorrow unless you're comfortable with cosmjs + the different messages. 

Btw could we take the package-json.lock change out of this PR, unless there's a specific reason we should downgrade to lockfile version 1?

**Comment by tombeynon** [comment]
cc @Galadrin for CRO - is that `1e8` needed here?

**Comment by evmosius** [comment]
@tombeynon Thanks for the review. Unfortunately I'm not particularly familiar with cosmjs + different messages. It would be greatly appreciated if you can implement the delegate message builder.

I've also nixed the `package-lock.json` change.

**Comment by tombeynon** [comment]
Very happy to, I'll update you here ASAP. Thanks again for the denom work 👌🏻

**Comment by tombeynon** [comment]
Sifchain is `1e18` for my notes. This info is available in the chain-registry assetlists

**Comment by Sdpierret** [comment]
> cc @Galadrin for CRO - is that `1e8` needed here?

yes, basecro is 8 decimal

**Comment by Sdpierret** [comment]
@evmosius  I made a change proposal on your branch ;)

File: src/networks.json [hunk 1]
```diff
@@ -769,7 +769,8 @@
   },
   {
     "name": "evmos",
-    "enabled": false,
+    "enabled": true,
+    "decimals": 1e18,
```
Comment (Galadrin) [inline src/networks.json:773]: can be fine to keep the decimal number from chain-registry

"decimals": 18

File: src/components/DelegateForm.js [hunk 2]
```diff
@@ -63,14 +63,16 @@ class DelegateForm extends React.Component {
...
+    const d = decimals ? decimals : 1e6;
     if(this.props.redelegate){
       messages.push({
         typeUrl: "/cosmos.staking.v1beta1.MsgBeginRedelegate",
         value: {
           delegatorAddress: address,
           validatorSrcAddress: this.props.validator.operator_address,
           validatorDstAddress: validatorAddress,
-          amount: coin(parseInt(parseFloat(amount) * 1000000), this.props.network.denom),
+          amount: coin(parseInt(parseFloat(amount) * d), this.props.network.denom),
```
Comment (Galadrin) [inline src/components/DelegateForm.js:75]: amount: coin(parseInt(parseFloat(amount) * math.pow(10, d), this.props.network.denom),

File: src/components/DelegateForm.js [hunk 3]
```diff
@@ -63,14 +63,16 @@ class DelegateForm extends React.Component {
     const address = this.props.address
     const validatorAddress = this.props.selectedValidator.operator_address
     let messages = []
+    const decimals = this.props.network.data.decimals
+    const d = decimals ? decimals : 1e6;
```
Comment (Galadrin) [inline src/components/DelegateForm.js:67]: const d = decimals ? decimals : 6;

File: src/components/Coins.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
 import _ from 'lodash'
 
 function Coins(props) {
-  function amount(coins){
-    return _.round(coins.amount / 1000000.0, 6)
+  function amount(coins, decimals){
+    if (!decimals) {
```
Comment (Galadrin) [inline src/components/Coins.js:5]: if (!decimals) {
      decimals = 6
    }
    return _.round(coins.amount / math.pow(10, decimals), 6)

**Review by Galadrin — CHANGES_REQUESTED** [review]

**Comment by tombeynon** [comment]
@Galadrin thank you for your suggestions and @evmosius hugely appreciate the initial work.

I'm going to merge the decimals fix in as-is and keep Evmos disabled for now, and work on the `ethermint.types.v1.EthAccount` bug in the linked issue. I'll get this sorted asap @evmosius

**Pull request merged**